src/material-experimental/mdc-form-field/_mdc-text-field-structure-overrides.scss

```diff
@@ -30,6 +30,13 @@
     height: auto;
   }
 
+  // Color inputs are a special case, because setting their height to
+  // `auto` will collapse them. The height value is an arbitrary number
+  // which was extracted from the user agent styles of Chrome and Firefox.
+  .mat-mdc-form-field-control[type='color'] {
+    height: 23px;
```

Comment:
Is there a way we can just unset the value by MDC?

Reply:
MDC isn't the problem, it actually has a fixed height already. A few lines up we override it to `auto` which breaks the `color` input.

Reply:
Ah right, that makes sense. I assume it would have also made sense to just not apply the `height: auto` for the color input, but at the same time I could see an argument that a static height ensures the form control always looks good.

Should we expand the comment a little and say that prefer an explicit height to make sure the control is working consistently?

Reply:
We could add `:not([type='color'])` instead, but that'll raise the specificity.

Reply:
The specificity seems not really low anyway on the `height: auto`, and I also wonder if the current specificity is actually enough to override `  .mdc-text-field--no-label:not(.mdc-text-field--textarea)
  .mat-mdc-form-field-control.mdc-text-field__input`? (did you check that?)